arm/implementations/joint_router.hpp

```diff
@@ -26,29 +39,34 @@ class joint_router
   {
 
     HAL_CHECK(
-      m_rotunda.position(static_cast<float>(p_arguments.rotunda_angle)));
+      m_rotunda_servo.position(static_cast<float>(p_arguments.rotunda_angle)));
 
     HAL_CHECK(
-      m_shoulder.position(static_cast<float>(-p_arguments.shoulder_angle)));
+      m_shoulder_servo.position(static_cast<float>(-p_arguments.shoulder_angle * shoulder_gear_ratio)));
 
-    HAL_CHECK(m_elbow.position(static_cast<float>(p_arguments.elbow_angle)));
+    HAL_CHECK(m_elbow_servo.position(static_cast<float>(p_arguments.elbow_angle * elbow_gear_ratio)));
 
     int left_wrist_angle = static_cast<float>(p_arguments.wrist_roll_angle) +
                            static_cast<float>(p_arguments.wrist_pitch_angle);
-    HAL_CHECK(m_left_wrist.position(static_cast<float>(left_wrist_angle)));
+    HAL_CHECK(m_left_wrist_servo.position(static_cast<float>(left_wrist_angle)));
```

Comment:
Feels like there are too many casts to float here. This could be converted into a function that provides this answer. But if you wanted to truncate this, it would be more expressive to use the `std::floor()` function on the float rather than implicitly casting it to `int` then back to `float`.